File: commands/fun/catfacts.go
package fun

import (
  "net/http"
  "encoding/json"
  "io/ioutil"
  "github.com/sapphire-cord/sapphire"
)

// Let me tell you a misterious cat fact.
// Aliases: catfact, kittenfact
// Cooldown: 3
func Catfacts(ctx *sapphire.CommandContext) {
  res, err := http.Get("https://catfact.ninja/fact")
  defer res.Body.Close()
  if err != nil {
    ctx.Reply("Error: %s", err)
    return
  }
  var data map[string]interface{}
  buf, err := ioutil.ReadAll(res.Body)
  if err != nil {
    ctx.Reply("Error: %s", err)
    return
  }
  err = json.Unmarshal(buf, &data)
  if err != nil {
    ctx.Reply("Error: %s", err)
    return
  }
  ctx.Reply("📢 **Catfact:** *%s*", data["fact"])
}

File: commands/fun/catfacts.js
const Command = require("../../structures/Command.js");
const fetch = require("node-fetch");

class CatFacts extends Command {
  constructor(...args) {
    super(...args, {
      aliases: ["catfact", "kittenfact"],
      cooldown: 3,
      description: "Let me tell you a misterious cat fact."
    });
    this.cost = 10;
  }
  
  async run(ctx) {
    const fact = await fetch("https://catfact.ninja/fact")
      .then((res) => res.json())
      .then(({ fact }) => fact);

    return ctx.reply(`📢 **Catfact:** *${fact}*`);
  }
}

module.exports = CatFacts;
